crates/ty_python_semantic/src/types/function.rs

```diff
@@ -713,17 +732,15 @@ impl<'db> FunctionType<'db> {
     /// Returns true if this method is decorated with `@classmethod`, or if it is implicitly a
     /// classmethod.
     pub(crate) fn is_classmethod(self, db: &'db dyn Db) -> bool {
-        self.has_known_decorator(db, FunctionDecorators::CLASSMETHOD)
-            || matches!(
-                self.name(db).as_str(),
-                "__init_subclass__" | "__class_getitem__"
-            )
+        self.iter_overloads_and_implementation(db)
```

Comment:
Nice! Glad to see we don't have to duplicate the source of truth for special-cased method names.